spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitTemplateTests.java

```diff
@@ -638,6 +642,9 @@ void resourcesClearedAfterTxFailsWithSync() throws IOException, TimeoutException
 		assertThatIllegalStateException()
 			.isThrownBy(() -> (TransactionSynchronizationManager.getSynchronizations()).isEmpty())
 			.withMessage("Transaction synchronization is not active");
+		assertThatExceptionOfType(AfterCompletionFailedException.class)
+			.isThrownBy(() -> ConnectionFactoryUtils.checkAfterCompletion());
+		ConnectionFactoryUtils.enableAfterCompletionFailureCapture(true);
```

Comment:
You probably mean `false` in the end of test.
Plus I believe you realize that this is global static flag, which can be changed in one thread, but checked in the other.
Is it really intentional?
Also: you probably have missed to mention this flag in the doc.

Reply:
Oops; yes; it was intentionally global - they shouldn't have to set it on each thread.

Doh - I had an unsaved change to the doc file.

Reply:
Good, but why this is still `true` in the end of this test ? 😄